pkg/apiserver/resthandler.go

```diff
@@ -488,8 +488,7 @@ func PatchResource(r rest.Patcher, scope RequestScope, typer runtime.ObjectTyper
 		gv := scope.Kind.GroupVersion()
 		codec := runtime.NewCodec(
 			scope.Serializer.EncoderForVersion(s, gv),
-			scope.Serializer.DecoderToVersion(s, unversioned.GroupVersion{Group: gv.Group, Version: runtime.APIVersionInternal}),
-		)
+			scope.Serializer.DecoderToVersion(s, gv))
```

Comment:
@liggit thanks! I've indeed added a lot of debug logging yesterday and learned this. One thing confused me a lot though, [this line](https://github.com/kubernetes/kubernetes/blob/ef0c9f0c5b8efbba948a0be2c98d9d2e32e0b68c/pkg/apiserver/apiserver_test.go#L134) causing a test case I wrote for alleged representation mixup to fail on pods. I tried removing it and converting resthandler patch tests to use v1.Pod instead of api.Pod on input (to make it more realistic) and everything seemed to work. So looks like I was right indeed and #25106 is actually just a problem in e2e test despite what @smarterclayton says.